src/Symfony/Component/DependencyInjection/Dumper/GraphvizDumper.php

```diff
@@ -166,7 +166,7 @@ private function findNodes()
         foreach ($container->getDefinitions() as $id => $definition) {
             $class = $definition->getClass();
 
-            if ('\\' === substr($class, 0, 1)) {
+            if ('\\' === $class[0]) {
```

Comment:
What if $class is empty? `substr` returns false if the offset is large than the length, the array access throws an error

https://3v4l.org/JIFTp
https://3v4l.org/TLVrU